#!/usr/bin/env python

# set up parameters that we care about
PACKAGE = 'jsk_interactive_marker'

from dynamic_reconfigure.parameter_generator_catkin import *;

from math import pi

gen = ParameterGenerator ()
orientation_enum = gen.enum([gen.const("INHERIT", int_t, 0, "INHERIT"),
                      gen.const("FIXED", int_t, 1, "FIXED"),
                             gen.const("VIEW_FACING", int_t, 2, "VIEW_FACING")],
                      "orientation")

gen.add("display_interactive_manipulator",   bool_t,   0, "Display Interactive rotate and translate manipulator",  True)
gen.add("display_interactive_manipulator_only_selected",   bool_t,   0, "Display interactive rotate and translate manipulator when selected.",  False)
gen.add("display_description_only_selected", bool_t, 0, "Display description when selected.", False)
gen.add("interactive_manipulator_orientation", int_t, 0, "interactive_manipulator_orientation", 0, 0, 2,
        edit_method = orientation_enum)

interaction_mode_enum = gen.enum([
    gen.const("NONE", int_t, 0, "NONE"),
    gen.const("MENU", int_t, 1, "MENU"),
    gen.const("BUTTON", int_t, 2, "BUTTON"),
    gen.const("MOVE_AXIS", int_t, 3, "MOVE_AXIS"),
    gen.const("MOVE_PLANE", int_t, 4, "MOVE_PLANE"),
    gen.const("ROTATE_AXIS", int_t, 5, "ROTATE_AXIS"),
    gen.const("MOVE_ROTATE", int_t, 6, "MOVE_ROTATE"),
    gen.const("MOVE_3D", int_t, 7, "MOVE_3D"),
    gen.const("ROTATE_3D", int_t, 8, "ROTATE_3D"),
    gen.const("MOVE_ROTATE_3D", int_t, 9, "MOVE_ROTATE_3D"),
], "interaction_mode")
gen.add("interaction_mode", int_t, 0, "interaction_mode", 7, 0, 9, edit_method=interaction_mode_enum)

exit (gen.generate (PACKAGE, "jsk_interactive_marker", "InteractiveSetting"))
